File: Pratica 3 - Estrutura de Repeticao/python_logic/desafio_MVP.py
produtos = []

while True:
    print("----MENU----")
    print("1 - Adicionar produto")
    print("2 - Listar produtos")
    print("3 - Sair")

    opcao = input("Escolha uma opção: ")

    if opcao == "1":
        nome_prod = input("Nome do produto: ")
        quant_est = int(input("Quantidade em estoque: "))
        preco_unit = float(input("Preço unitário: R$ "))

        if quant_est < 0:
            print("ERRO: A quantidade não pode ser um valor negativo. Por favor, tente novamente.")
        else:
            produto = {
                "nome": nome_prod,
                "quantidade": quant_est,
                "preco": preco_unit
            }

        produtos.append(produto)
        print("Produto cadastrado com sucesso!")

    elif opcao == "2":
        print("----ESTOQUE----")
        for i in produtos:
            print(f"Produto: {i['nome']} | Qtd: {i['quantidade']} | R$ {i['preco']:.2f}")

    elif opcao == "3":
        print("Saindo do sistema...")
        break

    else:
        print("Opção inválida!")

File: Pratica 3 - Estrutura de Repeticao/cpp_logic/desafio_MVP.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {

    vector<vector<string>> produtos;

    while (true) {
        cout << "\n----MENU----" << endl;
        cout << "1 - Adicionar produto" << endl;
        cout << "2 - Listar produtos" << endl;
        cout << "3 - Sair" << endl;

        string opcao;
        cout << "Escolha uma opção: ";
        cin >> opcao;

        if (opcao == "1") {
            string nome;
            int quant;
            float preco;

            cout << "Nome do produto: ";
            cin >> nome;

            cout << "Quantidade em estoque: ";
            cin >> quant;

            cout << "Preço unitário: R$ ";
            cin >> preco;

            if (quant < 0) {
                cout << "ERRO: quantidade não pode ser negativa." << endl;
            } else {
                vector<string> produto = {
                    nome,
                    to_string(quant),
                    to_string(preco)
                };

                produtos.push_back(produto);
                cout << "Produto cadastrado com sucesso!" << endl;
            }

        } else if (opcao == "2") {
            cout << "\n----ESTOQUE----" << endl;

            for (int i = 0; i < produtos.size(); i++) {
                cout << "Produto: " << produtos[i][0]
                     << " | Qtd: " << produtos[i][1]
                     << " | R$ " << produtos[i][2] << endl;
            }

        } else if (opcao == "3") {
            cout << "Saindo do sistema..." << endl;
            break;

        } else {
            cout << "Opção inválida!" << endl;
        }
    }

    return 0;
}